File: tests/lang-test/python/src/services/async_service.py
"""ripex-lang-test: Python async service — async/await, context manager."""
import asyncio
from typing import List, Literal


async def fetch_json(url: str) -> dict[str, str]:
    await asyncio.sleep(0)
    return {"url": url}


async def gather_all(urls: List[str]) -> List[dict[str, str]]:
    return await asyncio.gather(*(fetch_json(u) for u in urls))


class Resource:
    def __enter__(self) -> "Resource":
        return self

    def __exit__(self, *exc: object) -> Literal[False]:
        return False

File: tests/lang-test/csharp/Services/AsyncService.cs
// ripex-lang-test: C# async service — async/await, Task, cancellation.
using System;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;

namespace Ripex.Services
{
    public class AsyncService
    {
        public async Task<string> FetchJsonAsync(string url)
        {
            await Task.Delay(0);
            return url;
        }

        public async Task<List<string>> FetchAllAsync(IEnumerable<string> urls)
        {
            var tasks = new List<Task<string>>();
            foreach (var u in urls)
            {
                tasks.Add(FetchJsonAsync(u));
            }
            var results = await Task.WhenAll(tasks);
            return new List<string>(results);
        }

        public string WithTimeout(CancellationToken token)
        {
            return token.IsCancellationRequested ? "cancelled" : "ok";
        }
    }
}
